python_modules/libraries/dagster-snowflake/dagster_snowflake_tests/test_sql_component.py

```diff
@@ -189,3 +190,65 @@ def test_snowflake_sql_component_with_execution(snowflake_connect):
 
         # Verify the op description is set correctly
         assert asset_def.op.description == "This is a test op description"
+
+
+class RefreshExternalTableComponent(BaseSnowflakeSqlComponent):
+    """A custom component which refreshes an external table in Snowflake."""
+
+    table_name: str
+
+    @property
+    def sql_content(self) -> str:
+        return f"ALTER TABLE {self.table_name} REFRESH;"
+
+
+@mock.patch("snowflake.connector.connect", new_callable=create_mock_connector)
+def test_custom_snowflake_sql_component(snowflake_connect):
+    """Test that a custom SnowflakeSqlComponent subclass correctly builds and executes SQL."""
+    component_body = {
+        "type": "dagster_snowflake_tests.test_sql_component.RefreshExternalTableComponent",
+        "attributes": {
+            "table_name": "TESTDB.TESTSCHEMA.EXTERNAL_TABLE",
+            "assets": [{"key": "TESTDB/TESTSCHEMA/EXTERNAL_TABLE"}],
+        },
+    }
+    with scaffold_defs_sandbox(
+        component_cls=RefreshExternalTableComponent,
+    ) as defs_sandbox:
+        with defs_sandbox.load(component_body=component_body) as (component, defs):
+            defs_with_resource = Definitions.merge(
```

Comment:
there is a `with_resources` help method on definitions objects now for this use case